### PR TITLE
Fix mocking for `qml.qchem.mol_data` 

### PR DESCRIPTION
**Pull request by obliviateandsurrender**
**Context:** Fixes mocking for the `qml.qchem.mol_data` unit tests done using `patch`

**Description of the Change:** Method `get_cids` from the `pubchempy` module was not getting mocked correctly. We fix it by explicitly redefining it in the mocked variant of the module. 

**Benefits:** Unit tests will not have stochastic behaviour due to these unit tests

**Possible Drawbacks:** N/A

**Related GitHub Issues:** N/A

### CODE REVIEW

File: tests/qchem/test_structure.py [hunk 1]
```diff
@@ -402,9 +403,10 @@ def test_consistent_pubchem_mol_data(identifier, identifier_type):
     ref_mol_data_2d = (["H", "H"], np.array([[3.77945225, 0.0, 0.0], [5.66917837, 0.0, 0.0]]))
 
     mock_gc, mock_fc = ft.partial(mock_get_cids, pcp=pcp), ft.partial(mock_from_cid, pcp=pcp)
-    with patch.object(pcp.Compound, "get_cids", mock_gc) and patch.object(
+    with patch.dict(sys.modules, {"pubchempy": pcp}) and patch.object(
         pcp.Compound, "from_cid", mock_fc
     ):
+        pcp.get_cids = mock_gc
```
Comment (obliviateandsurrender) [inline tests/qchem/test_structure.py:409]: Did this because I couldn't patch `pcp.get_cids` directly using `with`.

**Review by rmoyard — APPROVED** [review]
Thanks just one comment :+1:

File: tests/qchem/test_structure.py [hunk 2]
```diff
@@ -314,26 +314,27 @@ def test_inconsistent_active_spaces(
...
+def mock_get_cids(identifier, namespace, pcp=None):
     """Return PubChem Compound ID for the provided identifier"""
     records = {
         297: [
             ("CH4", "name"),
-            ("74-82-8", "CAS"),
-            ("[C]", "SMILES"),
-            ("InChI=1S/CH4/h1H4", "InChI"),
-            ("VNWKTOKETHGBQD-UHFFFAOYSA-N", "InChIKey"),
+            ("74-82-8", "name"),
```
Comment (rmoyard) [inline tests/qchem/test_structure.py:322]: Is it really the name here and not `cas`?

Comment (obliviateandsurrender) [reply tests/qchem/test_structure.py:322]: We change it to `name` when using `CAS` [here](https://github.com/PennyLaneAI/pennylane/blob/1f93032a5f79ea0b9076d456b5b6c30515ef347d/pennylane/qchem/structure.py#L517).

**Pull request merged**